backend/libraries/ballerina-core/business rule engine/parser/expr.fs

```diff
@@ -407,6 +407,23 @@ module Expr =
                   let! case = JsonValue.AsString caseJson |> state.OfSum
                   let! value = Value.Parse valueJson
                   return Value.CaseCons(case, value)
+                }
+                state {
+                  let! fieldsJson = JsonValue.AsRecord json |> state.OfSum
+                  let! kindJson = fieldsJson |> sum.TryFindField "kind" |> state.OfSum
+
+                  do!
```

Comment:
common for everything -> could be a local utility fn

Reply:
Done